indigo_app/views/works.py

```diff
@@ -1535,3 +1536,58 @@ def get_context_data(self, **kwargs):
         )
         context_data["prefix"] = prefix
         return context_data
+
+
+class WorkFormConsolidationView(WorkViewBase, TemplateView):
+    template_name = "indigo_api/_work_form_consolidations_form.html"
+
+    def post(self, request, *args, **kwargs):
+        return self.get(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["formset"] = formset = ConsolidationsBaseFormset(self.request.POST,
+                                                    user=self.request.user,
+                                                    work=self.work,
+                                                    prefix="consolidations",
+                                                    form_kwargs={"work":self.work,
+                                                                 "user": self.request.user})
+
+        initial = []
+        if formset.is_valid():
+            for form in formset.forms:
+                delete = form.cleaned_data.get("DELETE")
+                if delete and not form.cleaned_data.get("id"):
+                    continue
+                initial.append({
+                    "date": form.cleaned_data["date"],
+                    "id": form.cleaned_data["id"],
+                    "DELETE": form.cleaned_data["DELETE"],
+                })
+
+            if self.request.POST.get("consolidation") == "add":
+                date = ""
+                if (
+                    self.work.principal and
+                    self.work.publication_date and
+                    self.place.settings.is_consolidation and
+                    self.place.settings.as_at_date and
```

Comment:
rather use self.work.as_at_date() as that takes the optional as_at_date_override on a work into account